server/schedule/region_scatterer.go

```diff
@@ -312,9 +330,23 @@ func (r *RegionScatterer) selectCandidates(region *core.RegionInfo, sourceStoreI
 	filters = append(filters, scoreGuard)
 	stores := r.cluster.GetStores()
 	candidates := make([]uint64, 0)
+	maxStoreTotalCount := uint64(0)
+	minStoreTotalCount := uint64(math.MaxUint64)
+	for _, store := range r.cluster.GetStores() {
+		count := context.selectedPeer.storeTotalCount(store.GetID())
+		if count > maxStoreTotalCount {
+			maxStoreTotalCount = count
+		}
+		if count < minStoreTotalCount {
+			minStoreTotalCount = count
+		}
+	}
 	for _, store := range stores {
-		if filter.Target(r.cluster.GetOpts(), store, filters) && !store.IsBusy() {
-			candidates = append(candidates, store.GetID())
+		storeCount := context.selectedPeer.storeTotalCount(store.GetID())
+		if storeCount < maxStoreTotalCount || storeCount == minStoreTotalCount {
```

Comment:
would you like to add more comment about it?

Reply:
updated.